src/bundle/Controller/User/UserForgotPasswordController.php

```diff
@@ -0,0 +1,185 @@
+{
+    /** @var FormFactory */
+    private $formFactory;
+
+    /** @var UserService */
+    private $userService;
+
+    /** @var \Swift_Mailer */
+    private $mailer;
+
+    /** @var \Twig_Environment */
+    private $twig;
+
+    /** @var string */
+    private $tokenTtl;
+
+    /**
+     * @param FormFactory $formFactory
+     * @param UserService $userService
+     * @param \Swift_Mailer $mailer
+     * @param \Twig_Environment $twig
+     * @param string $tokenTtl
+     */
+    public function __construct(
+        FormFactory $formFactory,
+        UserService $userService,
+        \Swift_Mailer $mailer,
+        \Twig_Environment $twig,
+        string $tokenTtl
+    ) {
+        $this->formFactory = $formFactory;
+        $this->userService = $userService;
+        $this->mailer = $mailer;
+        $this->twig = $twig;
+        $this->tokenTtl = $tokenTtl;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \Exception
+     * @throws InvalidOptionsException
+     * @throws \InvalidArgumentException
+     * @throws InvalidArgumentException
+     * @throws UnauthorizedException
+     * @throws NotFoundException
+     */
+    public function userForgotPasswordAction(Request $request): Response
+    {
+        $form = $this->formFactory->forgotUserPassword();
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+            $users = $this->userService->loadUsersByEmail($data->getEmail());
+
+            if (count($users) > 1) {
+                return $this->redirectToRoute('ezplatform.user.forgot_password.login');
+            }
+
+            if (!empty($users)) {
+                $user = reset($users);
+
+                $token = $this->updateUserToken($user);
+
+                $this->sendResetPasswordMessage($user->email, $token);
+            }
+
+            return $this->render('@EzPlatformAdminUi/Security/forgot_user_password_success.html.twig');
+        }
+
+        return $this->render('@EzPlatformAdminUi/Security/forgot_user_password.html.twig', [
+            'form_forgot_user_password' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \Exception
+     * @throws InvalidOptionsException
+     * @throws \InvalidArgumentException
+     * @throws InvalidArgumentException
+     * @throws UnauthorizedException
+     * @throws NotFoundException
+     */
+    public function userForgotPasswordLoginAction(Request $request): Response
+    {
+        $form = $this->formFactory->forgotUserPasswordWithLogin();
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+
+            try {
+                $user = $this->userService->loadUserByLogin($data->getLogin());
+            } catch (CoreNotFoundException $e) {
```

Comment:
`eZ\Publish\Core\Base\Exceptions\NotFoundException` extends `eZ\Publish\API\Repository\Exceptions\NotFoundException`. There's no need for aliasing `CoreNotFoundException`.